packages/dataviews/src/dataviews-layouts/list/index.tsx

```diff
@@ -210,96 +272,30 @@ function ListItem< Item >( {
 							width: 'auto',
 						} }
 					>
-						{ primaryAction && 'RenderModal' in primaryAction && (
-							<div role="gridcell">
-								<CompositeItem
-									store={ store }
-									render={
-										<Button
-											label={ primaryActionLabel }
-											icon={ primaryAction.icon }
-											isDestructive={
-												primaryAction.isDestructive
-											}
-											size="small"
-											onClick={ () =>
-												setIsModalOpen( true )
-											}
-										/>
-									}
-								>
-									{ isModalOpen && (
-										<ActionModal< Item >
-											action={ primaryAction }
-											items={ [ item ] }
-											closeModal={ () =>
-												setIsModalOpen( false )
-											}
-										/>
-									) }
-								</CompositeItem>
-							</div>
+						{ primaryAction && (
+							<PrimaryActionGridCell
+								idPrefix={ idPrefix }
+								primaryAction={ primaryAction }
+								item={ item }
+							/>
 						) }
-						{ primaryAction &&
-							! ( 'RenderModal' in primaryAction ) && (
-								<div role="gridcell" key={ primaryAction.id }>
-									<CompositeItem
-										store={ store }
-										render={
-											<Button
-												label={ primaryActionLabel }
-												icon={ primaryAction.icon }
-												isDestructive={
-													primaryAction.isDestructive
-												}
-												size="small"
-												onClick={ () => {
-													primaryAction.callback(
-														[ item ],
-														{ registry }
-													);
-												} }
-											/>
-										}
-									/>
-								</div>
-							) }
 						<div role="gridcell">
 							<DropdownMenu
 								trigger={
 									<CompositeItem
-										store={ store }
+										id={ generateDropdownTriggerCompositeId(
+											idPrefix
+										) }
 										render={
 											<Button
 												size="small"
 												icon={ moreVertical }
 												label={ __( 'Actions' ) }
 												accessibleWhenDisabled
 												disabled={ ! actions.length }
-												onKeyDown={ ( event: {
-													key: string;
-													preventDefault: () => void;
-												} ) => {
-													if (
-														event.key ===
-														'ArrowDown'
-													) {
-														// Prevent the default behaviour (open dropdown menu) and go down.
-														event.preventDefault();
-														store.move(
-															store.down()
-														);
-													}
-													if (
-														event.key === 'ArrowUp'
-													) {
-														// Prevent the default behavior (open dropdown menu) and go up.
-														event.preventDefault();
-														store.move(
-															store.up()
-														);
-													}
-												} }
+												onKeyDown={
```

Comment:
Moved the logic to the parent component, where the rest of the active composite ID logic is.